src/editors/Flow/view/Core/Graph/GraphBase.js

```diff
@@ -571,15 +573,47 @@ export default class GraphBase {
     this.warningsVisibility = isVisible;
   };
 
+  /**
+   * Update node running status
+   *
+   * @param {Object} nodes
+   * @param {*} robotStatus
+   */
   nodeStatusUpdated(nodes) {
+    // Let's wait untill allNodes are loaded to recall this function again
+    // What would happen sometimes is it was called before all nodes were loaded
+    // Causing some nodes to not light up when they should.
+    if (!this.allNodesLoaded)
+      return setTimeout(() => this.nodeStatusUpdated(nodes), 500);
```

Comment:
Is this the only to do this? Can we add a constant in the Timeout milliseconds?

Reply:
This is to prevent the nodes that aren't loaded yet, from not blinking ever. So we need to "listen" to the load confirmation of all nodes to re call this function with the same parameters it had before. Kinda like a debounce, but with a conditional.

Moved the timeout to a const. Nice catch :+1: